lib/shims.ml

```diff
@@ -0,0 +1,55 @@
+(* Each function in this file should be documented with the OCaml
+   version that introduced it, so we can clean this when we drop
+   support for older versions.
+ *)
+
+
+module Int_shims = struct
+  let equal (x : int) (y : int) =
+    (* 4.08 *)
+    x = y
+end
+
+module Seq_shims = struct
+  open! Seq
+
+  let rec append seq1 seq2 () =
+    (* 4.11 *)
+    match seq1 () with
+    | Nil -> seq2 ()
+    | Cons (x, next) -> Cons (x, append next seq2)
+
+  let rec flat_map f seq () =
+    (* 4.13 *)
+    match seq () with
+    | Seq.Nil -> Seq.Nil
+    | Cons (x, next) -> append (f x) (flat_map f next) ()
+
+end
+
+module List_shims = struct
+  open! List
+
+  let to_seq list =
+    (* 4.07 *)
+    let rec aux l () = match l with
+      | [] -> Seq.Nil
+      | x :: tail -> Seq.Cons (x, aux tail)
+    in
+    aux list
+end
+
+module Array_shims = struct
+  open! Array
+
+  let to_seq a =
+    (* 4.07 *)
+    let rec aux i () =
+      if i < Array.length a
+      then
+        let x = Array.unsafe_get a i in
+        Seq.Cons (x, aux (i + 1))
+      else Seq.Nil
+    in
+    aux 0
+end
```

Comment:
Could you inline this module int `Hamt`? I'm planning to vendor this library in another project, and it would be nice if it was just one file.